### PR TITLE
feat(pipeline executions/deck): Add support for max concurrent pipeli…

### PR DESCRIPTION
**Pull request by victor-homedepot**
…ne executions

It adds support for max concurrent pipeline executions. If concurrent pipeline execution is enabled, pipelines will queue when the max concurrent pipeline executions is reached. Any queued pipelines will be allowed to run once the number of running pipeline executions drops below the max. If the max is set to 0, then pipelines will not queue.

This is the proposed change for spinnaker/spinnaker#6558.

Changes to orca, front50 and echo are also needed in order to expose max concurrent pipeline executions as a pipeline configuration to Spinnaker users.

[Orca PR](https://github.com/spinnaker/orca/pull/4193)
[Front50 PR](https://github.com/spinnaker/front50/pull/1080)
[Echo PR](https://github.com/spinnaker/echo/pull/1134)

** Discussion was had with SIG team regarding this feature.

<details>
<summary>Feature Screenshots</summary>

### Enabled  
Screen Shot 2021-11-30 at 4 44 28 PM

### Tooltip
Screen Shot 2021-11-30 at 4 44 09 PM

### Disabled
Screen Shot 2021-11-30 at 4 44 53 PM

### CODE REVIEW

File: packages/core/src/help/helpContents.registry.spec.ts [hunk 1]
```diff
@@ -21,4 +21,12 @@ describe('Help contents registry', () => {
...
     });
   });
+
+  describe('max concurrent definition', () => {
+    it('provides the expected definition for max concurrent', () => {
+      const definition =
+        '<p>If concurrent pipeline execution is enabled, this variable sets the maximum number of concurrent pipelines executing. <br><br>If set to 0, then max is unlimited.</p>';
+      expect(HelpContentsRegistry.getHelpField('pipeline.config.parallel.max.concurrent')).toEqual(definition);
+    });
+  });
```
Comment (christopherthielen) [inline packages/core/src/help/helpContents.registry.spec.ts:31]: I don't think this is a useful test

**Review by christopherthielen — APPROVED** [review]

**Comment by guido9j** [comment]
@christopherthielen : can this be merged to master? Thanks

**Comment by aimeeu** [comment]
@victor-homedepot  Hi Victor -  I don't see any documentation for this beyond what's in this PR. Is there any configuration that has to be done to Spinnaker to enable this new feature? 
For example, if I uncheck the "Disable concurrent pipeline executions.." box but the "Maximum concurrent pipeline executions " doesn't appear, what would cause that?

**Comment by guido9j** [comment]
@aimeeu: No configuration in Spinnaker is needed to enable feature. This feature will be part of Spinnaker 1.28.x, which is not release yet (the labels in the pull request shows the target Spinnaker release).

**Pull request merged**